### PR TITLE
Improve clock handling logic for gameplay

### LINKED ISSUE #1821: Fail- and PauseOverlay can be open at the same time
I somehow managed to pause the second I failed a song and both overlays opened at the same time.
qq

### PR DESCRIPTION
**Pull request by peppy**
Resolves #1821 a second time.

- [x] Depends on ppy/osu-framework#1423

Also cleans up the logic and refactors variable names to be a bit more understandable.

### CODE REVIEW

File: osu.Game/Screens/Play/PauseContainer.cs [hunk 1]
```diff
@@ -122,6 +120,9 @@ protected override void Update()
             if (!game.IsActive && CanPause)
                 Pause();
 
+            if (!IsPaused)
+                framedClock.ProcessFrame();
```
Comment (smoogipoo) [inline osu.Game/Screens/Play/PauseContainer.cs]: I don't like that we're processing what looks like the gameplay clock(?) in the pause container. I would rather that this was done at a `Player` level.

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Screens/Play/PauseContainer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Play/PauseContainer.cs]: Hmm, the logic here was that only things which are children of the pause container use this clock. I was hoping eventually we wouldn't need to pass it to anything but the pause container.

The `PauseContainer` was supposed to move all pause/go logic out of `Player` itself

**Comment by smoogipoo** [comment]
Wanna update the framework once more?

**Comment by peppy** [comment]
It's already up-to-date.

**Review by smoogipoo — APPROVED** [review]
Seems ok.

**Pull request merged**